Add =~ and !~ operators and allow strings

jcubic opened the pull request:
I've added perl =~ and !~ operators and now you can compare agains strings, regexes need to be in quotes like in php but this can be changed.

You can merge but you need to resolve the conflict in your changes.

jcubic committed:
Add =~ and !~ operators and allow strings

jcubic committed:
Fix floats with e

jcubic committed:
Remove debug code

optimistex commented:
Привет!
Спасибо за информацию о доработках. Влил твои правки в этот репозиторий.

У меня к тебе 2 просьбы:

1. Отдать дань уважения Miles Kaufmann, и отказаться от ребрендинга. То есть сохранить оригинальное название файла "evalmath.class.php" и оригинальные названия классов "EvalMath", "EvalMathStack". Пусть это будет маленькой библиотекой )))
Для работы с формулами предлагаю использовать "Expression.php >> \optimistex\expression\Expression" (пространство имен можно поменять, предложишь лучше!?).

2. Дай несколько примеров работы со строками, с учетом твоих доработок, что бы дополнить справку в файле "README.md", или дополни сам если незатруднит.

jcubic commented:
I've added Unit Tests and Makefile to run them, you can check and see my =~ operators and strings comparators in action in [tests/Expression.php](https://github.com/jcubic/expression.php/blob/master/tests/Expression.php)

I've changed the class name because when I've added strings it's no longer EvalMath but arbitrary expressions evaluator.

optimistex commented:
Залил твой код к себе в проект.

Исправлено:
1. Отформатировал код.
2. Убрал неиспользуемую переменную evalmath.class.php/tests/Expression.php:118
3. Переименовал метод \ExpressionStack::empty в \ExpressionStack::isEmpty, т.к. получил предупреждение об использовании зарезервированного слова empty()
